configure.ac

```diff
@@ -38,6 +38,14 @@ AC_MSG_RESULT([$ql_version])
 
 # check for tools
 AC_PATH_PROG([SWIG], [swig])
+if test "x$SWIG" != x; then
+	SWIG_VERSION=`swig -version 2>&1|grep Version|cut -d" " -f3`
+	SWIG_MAJOR=${SWIG_VERSION%%.*}
+	if test ${SWIG_MAJOR} = "2"; then
+		AC_MSG_WARN([You have SWIG $SWIG_VERSION installed, but at least SWIG 3.0.1 is recommended. Some features may not work. ])
+		#SWIG=
```

Comment:
When uncommented, this sets `SWIG` to null so that the outdated version won't be used.

I'm not sure what to do here. What happens if you run an outdated SWIG on the sources? Do we get something usable but with less features, or something that doesn't work?

Reply:
As I wrote, I am not an expert in that space. I found this issue while compiling on Amazon AWS machine running RedHat EL. By default there was a swig with major version 2 on the system. The compiled QuantLib exported to python was very poor, I guess less than 50% of functionality. If I remember correctly, most of the functions/ classes were smart pointer was in place were not available.
Installing recent swig solved the issue. I think that we don't have to solve the issue directly. If someone installs quantlib python with old swig and finds very small functionality, I guess most of the cases will try to recompile and will see the warning (that has missed during the first installation). Then he will find his way to upgrade swig...